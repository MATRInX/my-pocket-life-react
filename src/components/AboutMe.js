import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { LINKS } from '../fixtures/links';
import selectAboutMePage from '../selectors/aboutMePage';

export const AboutMe = ({ aboutMePage }) => (
  <div>
    {
      aboutMePage === undefined ? (
        <div>Loading...</div>
      ) : (
        <div>
          <h2>Kilka słów o mnie</h2>
          <Link to={LINKS.ABOUT_ME}>
            <img src={aboutMePage.image}/>
          </Link>
          <p>
            Mam na imię Aneta i od półtora roku jestem mamą. Wraz z mężem i synem mieszkamy w Tarnowskich Górach.
          </p>
          <Link to={LINKS.ABOUT_ME}>Czytaj więcej...</Link>
        </div>
      )
    }    
  </div>
);

const mapStateToProps = ({ pages } = state) => ({
  aboutMePage: selectAboutMePage(pages)
});

export default connect(mapStateToProps)(AboutMe);