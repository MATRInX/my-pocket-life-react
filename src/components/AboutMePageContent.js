import React from 'react';

const AboutMePageContent = () => (
  <div>
    AboutMePageContent content...
  </div>
);

export default AboutMePageContent;