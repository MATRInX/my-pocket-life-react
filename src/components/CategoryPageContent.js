import React from 'react';

const CategoryPageContent = (props) => (
  <div>
    CategoryPageContent - category:
    <p>{props.match.params.category}</p>
  </div>
);

export default CategoryPageContent;