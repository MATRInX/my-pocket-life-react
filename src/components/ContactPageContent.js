import React from 'react';

const ContactPageContent = () => (
  <div>
    ContactPageContent content...
  </div>
);

export default ContactPageContent;