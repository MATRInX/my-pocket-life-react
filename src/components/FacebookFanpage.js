import React from 'react';

const FacebookFanpage = () => (
  <div>FacebookFanpage content</div>
);

export default FacebookFanpage;