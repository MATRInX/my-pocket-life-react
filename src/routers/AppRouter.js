import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import createHistory from 'history/createBrowserHistory';
import AboutMePageContent from '../components/AboutMePageContent';
import ContactPageContent from '../components/ContactPageContent';
import CategoryPageContent from '../components/CategoryPageContent';
import Footer from '../components/Footer';
import Header from '../components/Header';
import MainPageContent from '../components/MainPageContent';
import SearchPageContent from '../components/SearchPageContent';
import SubHeader from '../components/SubHeader';

export const history = createHistory();

const AppRouter = () => (
  <Router history={history}>
    <div>
    <Header />
    <SubHeader />
    <Switch>
      <Route path="/" component={MainPageContent} exact={true}/>
      <Route path="/o-mnie" component={AboutMePageContent} />
      <Route path="/kontakt" component={ContactPageContent} />
      <Route path="/szukaj/:queryString" component={SearchPageContent} />
      <Route path="/kategoria/:category" component={CategoryPageContent} />
    </Switch>
    <Footer />
    </div>
  </Router>
);

export default AppRouter;