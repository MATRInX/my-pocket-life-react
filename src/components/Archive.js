import React from 'react';
import { connect } from 'react-redux';
import selectArchiveCategory from '../selectors/archiveCategory';
import { setArchiveMonth, setArchiveYear } from '../actions/filters';
import { history } from '../routers/AppRouter';

export class Archive extends React.Component {
  state = {
    selectValue: this.props.selectValue ? this.props.selectValue : 
      this.props.filters.archiveMonth + '|' + this.props.filters.archiveYear
  };

  componentDidMount() {
    if (this.state.selectValue !== '0|0') {
      const filters = this.state.selectValue.split('|');
      this.props.setArchiveYear(Number(filters[1]));
      this.props.setArchiveMonth(Number(filters[0]));
    }
  }

  onSelectChange = e => {
    const selectValue = e.target.value;
    this.setState(() => ({ selectValue }));
    const filters = selectValue.split('|');
    this.props.setArchiveYear(Number(filters[1]));
    this.props.setArchiveMonth(Number(filters[0]));
    history.push(`/archiwum/${filters[1]}/${filters[0]}`);
  }

  render() {
    return (
      <div>
        <p>Archive content: {this.state.selectValue}</p>
        <select 
          value={this.state.selectValue}
          onChange={this.onSelectChange}
        >
          <option value="0|0">Wybierz miesiÄ…c...</option>
          {
            this.props.archiveData.map((archiveItem, index) => {
              return (
                <option 
                  key={index} 
                  value={archiveItem.month+'|'+archiveItem.year}
                >
                  {archiveItem.text}  
                </option>
              )
            })
          }
        </select>
      </div>
    );
  }
};

const mapStateToProps = ({ posts, filters } = state) => ({
  archiveData: selectArchiveCategory(posts),
  filters
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  setArchiveMonth: month => dispatch(setArchiveMonth(month)),
  setArchiveYear: year => dispatch(setArchiveYear(year))
});

export default connect(mapStateToProps, mapDispatchToProps)(Archive);