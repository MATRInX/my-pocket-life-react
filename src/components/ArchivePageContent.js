import React from 'react';
import Archive from './Archive';
import PostList from './PostsList';

const ArchivePageContent = ({ match }) => (
  <div>
    ArchivePageContent content...
    <PostList />
    <Archive selectValue={match.params.month +'|'+match.params.year}/>
  </div>
);

export default ArchivePageContent;