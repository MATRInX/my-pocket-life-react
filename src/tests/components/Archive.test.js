import React from 'react';
import { shallow } from 'enzyme';
import { Archive } from '../../components/Archive';
import { dummyPosts } from '../../fixtures/dummyData';
import { history } from '../../routers/AppRouter';

test('should correctly render Archive widget on sidebar', () => {
  const wrapper = shallow(<Archive archiveData={dummyPosts}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render Archive widget on sidebar with no archive data', () => {
  const wrapper = shallow(<Archive archiveData={[]}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly set default selectValue', () => {
  const defaultSelectValue = '0|0';
  const wrapper = shallow(<Archive archiveData={[]}/>);
  expect(wrapper.state('selectValue')).toBe(defaultSelectValue);
});

test('should correctly set selectValue if it is different than filters value', () => {
  const selectValue = '4|2016';
  const filtersValue = '5|2017';
  const filters = { filtersValue };
  const setPostsFilterSpy = jest.fn();
  const wrapper = shallow(<Archive 
    archiveData={dummyPosts} 
    filters={filters} 
    selectValue={selectValue}
    setPostsFilter={setPostsFilterSpy}
    />);
  expect(setPostsFilterSpy).toHaveBeenCalledWith(selectValue);
});

test('should set properly location after select archive month', () => {
  const month = '4';
  const year = '2018';
  const value = `${month}|${year}`;
  const url = `/archiwum/${year}/${month}`;
  const setPostsFilterSpy = jest.fn();
  const wrapper = shallow(<Archive archiveData={dummyPosts} setPostsFilter={setPostsFilterSpy}/>);
  wrapper.find('select').simulate('change', {
    target : { value }
  });
  expect(wrapper.state('selectValue')).toBe(value);
  expect(history.location.pathname).toBe(url);
  expect(setPostsFilterSpy).toHaveBeenCalledWith(value);
});