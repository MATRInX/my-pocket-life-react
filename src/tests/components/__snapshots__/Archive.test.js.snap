// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Archive widget on sidebar 1`] = `
<div>
  <p>
    Archive content: 
    0|0
  </p>
  <select
    onChange={[Function]}
    value="0|0"
  >
    <option
      value="0|0"
    >
      Wybierz miesiąc...
    </option>
    <option
      key="0"
      value="undefined|undefined"
    />
    <option
      key="1"
      value="undefined|undefined"
    />
    <option
      key="2"
      value="undefined|undefined"
    />
  </select>
</div>
`;

exports[`should correctly render Archive widget on sidebar with no archive data 1`] = `
<div>
  <p>
    Archive content: 
    0|0
  </p>
  <select
    onChange={[Function]}
    value="0|0"
  >
    <option
      value="0|0"
    >
      Wybierz miesiąc...
    </option>
  </select>
</div>
`;
